python/ray/train/v2/trainer.py

```diff
@@ -0,0 +1,327 @@
+import abc
```

Comment:
Train is still in alpha. Instead of having a "v2", can we move the old classes to "old" and make this the main interface?

We should de-clutter the train directory as much as possible.